### DISCUSSION TITLE
Cargo build failed: E0599, E0658

### DISCUSSION THREAD

**Discussion by josiahking** [rusqlite/rusqlite#1101] [Questions] [answered]
I got this error while trying to my program with rusqlite.
error[E0658]: use of unstable library feature 'maybe_uninit' (see issue #53491)
 --> \.cargo\registry\src\github.com-1ecc6299db9ec823\rusqlite-0.21.0\src\inner_connection.rs:2:5
  |
2 | use std::mem::MaybeUninit;
  |     ^^^^^^^^^^^^^^^^^^^^^

error[E0658]: use of unstable library feature 'maybe_uninit' (see issue #53491)
  --> \.cargo\registry\src\github.com-1ecc6299db9ec823\rusqlite-0.21.0\src\inner_connection.rs:83:26
   |
83 |             let mut db = MaybeUninit::uninit();
   |                          ^^^^^^^^^^^^^^^^^^^

error[E0658]: use of unstable library feature 'maybe_uninit' (see issue #53491)
   --> \.cargo\registry\src\github.com-1ecc6299db9ec823\rusqlite-0.21.0\src\inner_connection.rs:232:26
    |
232 |         let mut c_stmt = MaybeUninit::uninit();
    |                          ^^^^^^^^^^^^^^^^^^^

error[E0658]: use of unstable library feature 'maybe_uninit' (see issue #53491)
   --> \.cargo\registry\src\github.com-1ecc6299db9ec823\rusqlite-0.21.0\src\inner_connection.rs:234:26
    |
234 |         let mut c_tail = MaybeUninit::uninit();
    |                          ^^^^^^^^^^^^^^^^^^^

error[E0599]: no function or associated item named `uninit` found for type `std::mem::MaybeUninit<_>` in the current scope
  --> \.cargo\registry\src\github.com-1ecc6299db9ec823\rusqlite-0.21.0\src\inner_connection.rs:83:26
   |
83 |             let mut db = MaybeUninit::uninit();
   |                          ^^^^^^^^^^^^^^^^^^^ function or associated item not found in `std::mem::MaybeUninit<_>`

error[E0599]: no function or associated item named `uninit` found for type `std::mem::MaybeUninit<_>` in the current scope
   --> \.cargo\registry\src\github.com-1ecc6299db9ec823\rusqlite-0.21.0\src\inner_connection.rs:232:26
    |
232 |         let mut c_stmt = MaybeUninit::uninit();
    |                          ^^^^^^^^^^^^^^^^^^^ function or associated item not found in `std::mem::MaybeUninit<_>`

error[E0599]: no function or associated item named `uninit` found for type `std::mem::MaybeUninit<_>` in the current scope
   --> \.cargo\registry\src\github.com-1ecc6299db9ec823\rusqlite-0.21.0\src\inner_connection.rs:234:26
    |
234 |         let mut c_tail = MaybeUninit::uninit();
    |                          ^^^^^^^^^^^^^^^^^^^ function or associated item not found in `std::mem::MaybeUninit<_>`

error[E0658]: use of unstable library feature 'transpose_result' (see issue #47338)
   --> \.cargo\registry\src\github.com-1ecc6299db9ec823\rusqlite-0.21.0\src\row.rs:113:14
    |
113 |             .transpose()
    |              ^^^^^^^^^

error[E0658]: use of unstable library feature 'transpose_result' (see issue #47338)
   --> \.cargo\registry\src\github.com-1ecc6299db9ec823\rusqlite-0.21.0\src\row.rs:145:14
    |
145 |             .transpose()
    |              ^^^^^^^^^

error: aborting due to 9 previous errors

Some errors occurred: E0599, E0658.
For more information about an error, try `rustc --explain E0599`.
error: Could not compile `rusqlite`.
warning: build failed, waiting for other jobs to finish...
error: build failed

**Comment by gwenn** [comment, marked as answer]
See https://doc.rust-lang.org/std/mem/union.MaybeUninit.html
`1.36.0` means that `MaybeUninit` has been stabilized in version 1.36.
And the errors mean that your rust version is less than 1.36.
Idem for https://doc.rust-lang.org/std/result/enum.Result.html#method.transpose.
